server/src/test/java/org/elasticsearch/index/mapper/ContentPathTests.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class ContentPathTests extends ESTestCase {
+
+    public void testAddPath() {
+        ContentPath contentPath = new ContentPath();
+        contentPath.add("foo");
+        contentPath.add("bar");
+        contentPath.add("baz");
+        String[] path = contentPath.getPath();
+        assertEquals(path[0], "foo");
+        assertEquals(path[1], "bar");
+        assertEquals(path[2], "baz");
+    }
+
+    public void testRemovePath() {
+        ContentPath contentPath = new ContentPath();
+        contentPath.add("foo");
+        String[] path = contentPath.getPath();
+        assertEquals(path[0], "foo");
+        contentPath.remove();
+        assertNull(path[0]);
+    }
+
+    public void testRemovePathException() {
+        ContentPath contentPath = new ContentPath();
+        contentPath.add("foo");
+        contentPath.remove();
+        expectThrows(IndexOutOfBoundsException.class, () -> contentPath.remove());
+    }
+
+    public void testExpandpath() {
```

Comment:
I don't think we should have this test. It sort of relies on an internal implementation detail of how we expand the internal array at a given length.

This test should ensure that the `10` is correctly added to the path, that `path.pathAsText` is correct, and ` contentPath.length()` is correct.

Reply:
Done! Thanks for the feedback!